src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -34,43 +37,72 @@
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
-        }
+        final Matcher dynamicMatcher = DYNAMIC_COMMAND_FORMAT.matcher(userInput.trim());
+
+        if (dynamicMatcher.matches() && dynamicMatcher.group("inputType").equals("dynamic")) {
+            final String arguments = dynamicMatcher.group("arguments").trim();
+            String[] splitArgs = arguments.split(" ");
+
+            if (isCommand(arguments, splitArgs)) {
+                commandBox.setDynamicAsFalse();
+                return new ListCommand();
+            } else {
+                commandBox.setDynamicAsTrue();
+                return new FindCommandParser().parse(arguments);
+            }
+        } else {
+            if (!matcher.matches()) {
+                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, HelpCommand.MESSAGE_USAGE));
+            }
+
+            final String commandWord = matcher.group("commandWord");
+            final String arguments = matcher.group("arguments");
 
-        final String commandWord = matcher.group("commandWord");
-        final String arguments = matcher.group("arguments");
-        switch (commandWord) {
+            switch (commandWord) {
 
-        case AddCommand.COMMAND_WORD:
-            return new AddCommandParser().parse(arguments);
+            case AddCommand.COMMAND_WORD:
+                return new AddCommandParser().parse(arguments);
 
-        case EditCommand.COMMAND_WORD:
-            return new EditCommandParser().parse(arguments);
+            case EditCommand.COMMAND_WORD:
+                return new EditCommandParser().parse(arguments);
 
-        case DeleteCommand.COMMAND_WORD:
-            return new DeleteCommandParser().parse(arguments);
+            case DeleteCommand.COMMAND_WORD:
+                return new DeleteCommandParser().parse(arguments);
 
-        case ClearCommand.COMMAND_WORD:
-            return new ClearCommand();
+            case ClearCommand.COMMAND_WORD:
+                return new ClearCommand();
 
-        case FindCommand.COMMAND_WORD:
-            return new FindCommandParser().parse(arguments);
+            case FindCommand.COMMAND_WORD:
+                return new FindCommandParser().parse(arguments);
 
-        case ListCommand.COMMAND_WORD:
-            return new ListCommand();
+            case ListCommand.COMMAND_WORD:
+                return new ListCommand();
 
-        case ExitCommand.COMMAND_WORD:
-            return new ExitCommand();
+            case ExitCommand.COMMAND_WORD:
+                return new ExitCommand();
 
-        case HelpCommand.COMMAND_WORD:
-            return new HelpCommand();
+            case HelpCommand.COMMAND_WORD:
+                return new HelpCommand();
 
-        default:
-            throw new ParseException(MESSAGE_UNKNOWN_COMMAND);
+            default:
+                throw new ParseException(MESSAGE_UNKNOWN_COMMAND);
```

Comment:
Actually, can we call the dynamic find in the default case? Then we don't need the isCommand() method?

Reply:
Yes I agree with the case that we won't have any unknown commands in the case of dynamic inputs. However, I think the error arises when the user enters an unknown command, this will then not give rise to any error handling. Do you think this is possible?